content/ytt/docs/latest/lang-ref-ytt-schema.md

```diff
@@ -0,0 +1,226 @@
+
+- `system_domain` is an empty string, by default.
+- `load_balancer.enable` is true by default.
+- `databases[].adapter` is `"postgres"`, by default.
+
+### Inferring Defaults for Mappings
+
+The set of items of a map are its default: any item missing will be automatically inserted with _its_ defaults.
+
+- if `load_balancer` is omitted from supplied Data Values, it is inserted:
+    - `load_balancer.enabled` is `true`
+    - `load_balancer.static_ip` is `""`
+- if `load_balancer` is specified with only `static_ip`, `enabled` is `true`.
+
+When the Configuration Consumer supplies an empty map (i.e. `{}`), all items are defaulted:
+
+For example,
+
+`load_balancer: {}`
+
+is defaulted to:
+
+```yaml
+load_balancer:
+  enable: true
+  static_ip: ""
+```
+
+### Inferring Defaults for Arrays
+
+The default value for all arrays is an empty array. Arrays are the only type for which defaults are **not** inferred.
+
+From the example, above,
+
+- `app_domains`, if not otherwise specified is `[]`
+- `databases`, if not otherwise specified is `[]`
+
+When values for an array _are_ provided by the Configuration Consumer, each item is defaulted based on the type specified in the schema.
+
+Focusing on just `databases`, if the Configuration Consumer supplies this Data Value:
+
+```yaml
+#@data/values
+---
+databases:
+- name: uaa
+- name: capi
+  host: capi-db.svc.cluster.local
+  secretRef:
+    name: capi-db-credentials
+- {}
+```
+
+The resulting Data Values (i.e. that are supplied to templates), would be:
+
+```yaml
+databases:
+- name: uaa
+  adapter: postgresql
+  host: ""
+  port: 5432
+  user: admin
+  secretRef:
+    name: ""
+- name: capi
+  adapter: postgresql
+  host: capi-db.svc.cluster.local
+  port: 5432
+  user: admin
+  secretRef:
+    name: capi-db-credentials
+- name: ""
+  adapter: postgresql
+  host: ""
+  port: 5432
+  user: admin
+  secretRef:
+    name: ""
+```
+
+## Defining Schema Explicitly
+
+Configuration Authors may override inferred typing and defaults through annotations.
+
+### @schema/type
+
+Explicitly sets the type of the annotated node.
+
+```yaml
+@schema/type any=True
+```
+
+- `any` (`bool`) — whether or not any and all types are permitted on this node (and its children)
+    - to avoid confusion, this keyword argument is mutually exclusive to all other arguments; if this argument is
+      provided along with others, an error results.
+__
+
+### @schema/nullable
+
+Extends the type of the node to include "null" _and_ to be "null" by default.
+
+```yaml
+@schema/nullable
+```
+
+
+where `name` can hold a string, but also be null. `name` is `null` by default.
+
+_Example: Nullable map and string_ 
+
+```yaml
+#@schema/nullable
+aws:
+  username: ""
+  password: ""
+
+#@schema/nullable
+name: ""
+```
+
+where `aws` and `name` are null by default. When set in a data value file `aws` will be a map with the two items, and `name` will be a string.
```

Comment:
I think the example of having _no_ data values might go over well as the first part of the example... then elaborate with a concrete Data Value file and show how `aws:`, if any non-null value is given, the rest are defaulted.

Reply:
great idea, I updated the example. Let me know if feedback